### PR TITLE
Update auth with metadata and new way of managing roles #24

### PR DESCRIPTION
**Pull request by mateuszkrasucki**
- UPD auth backend (storage, services, commands)
    - UPD auth API docs with new definition of user and validation pair;
      added definition for location, organization, clinic objects
    - ADD local & cloud tags to auth API schema
    - ADD location, organization, clinic & userRoles storage
    - ADD userRoles implementation & domain in casbin
      - Relationships between users, entities (organization, clinic etc.)
        and roles are now defined through userRole entities. As a result
        roles are assigned to user within domain of entity:
          - organization
          - clinic
          - location
          - user (for controllin personal files modification rights)
      - UserRoles can also be assigned globally (apply everywehre)
        or with wildcard for type of domain (e.g. all clinics).
      - UserRoles can be searched based on all their parameters.
      - As a result domainType and domainID is now part of ValidationPair
        entity as it needs to be specified in what domain user needs to have
        specific resource right.
      - Predefined memberRole added to for users membership in organisation
        or clinic without additional rights.
      - Removed indication of supporting query parameters in auth API for search
        where they are not really supported.
      - Updates initial data loading mechanism.
    - UPD user, role, rule storage to reflect other changes
    - UPD all the updates in auth storage are now
      executed within one bolt transaction
    - UPD accountManager service was renamed to authDataManager service
      and new functionality was added related to location, organization,
      clinic and new domain-based validation.
    - UPD move all auth handlers to authDataManager
    - ADD users 'me' get endpoints to auth API
    - ADD view authDataManager handlers to localAuth
    - ADD ability to provision auth storage with yaml
      - Adds ability to provision cloudAuth storage with data in yaml file.
        Allows easy testing and visibility of initial data for development.
      - Adds documentation.
    - FIX storage sync consumer tests
      - Another adjustment made in sync storage consumer test to ensure
        reproducable result. Wait was added before service is cleaned in
        durability test.
    - ADD domainType and domainID as environment var
      - Adds domainType and domainID as common environment variable
        taken by components.
      - Updates authorizer and authetnicator to take into cosideration both
        values.
      - Updates initial data for auth storages.
- UPD auth cloud dashboard to support management of updated backend
- Related: #24

### CODE REVIEW

File: storage/auth/casbin.go [hunk 1]
```diff
@@ -106,21 +175,20 @@ func BinaryMatchFunc(args ...interface{}) (interface{}, error) {
 	return (bool)(binaryMatch(val1, val2)), nil
 }
 
-// SelfMatch replaces {self} in key2 with subject and then compares it with key1
+// SelfReplace replaces {self} in key2 with subject and then returns it for comparison.
```
Comment (dominikznidar) [inline storage/auth/casbin.go]: Function is still called `SelfMatch`

File: sync/storage/gomock_reflect_347889631/prog.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			os.Exit(1)
+		}
+		intf.Name = it.sym
+		pkg.Interfaces = append(pkg.Interfaces, intf)
+	}
+	if err := gob.NewEncoder(os.Stdout).Encode(pkg); err != nil {
+		fmt.Fprintf(os.Stderr, "gob encode: %v\n", err)
+		os.Exit(1)
+	}
+}
```
Comment (dominikznidar) [inline sync/storage/gomock_reflect_347889631/prog.go]: Do we need these reflect files?

Comment (mateuszkrasucki) [reply sync/storage/gomock_reflect_347889631/prog.go]: No, will remove.

File: storage/auth/casbin.go (hunk 1 above)
Comment (mateuszkrasucki) [reply storage/auth/casbin.go]: Thx.

File: cmd/waitlist/main.go [hunk 3]
```diff
@@ -38,7 +38,7 @@ func main() {
 		logger.Fatal().Err(err).Msg("Failed to initialize waitlist storage")
 	}
 
-	auth := authorizer.New("https://localAuth/auth/validate", logger)
+	auth := authorizer.New("clinic", "e4ebb41b-7c62-4db7-9e1c-f47058b96dd0", "https://localAuth/auth/validate", logger)
```
[the remark this answers is not in the document]
Comment (mateuszkrasucki) [reply cmd/waitlist/main.go]: Yeah, hardcoded it just for time being and forgot. Will fix.

File: cmd/localAuth/main.go [hunk 4]
```diff
@@ -103,29 +108,88 @@ func main() {
...
+			"rules",
+			"database",
+		}))
+
+	// set handler with middlewares
 	handler := cors.New(cors.Options{
-		AllowedMethods: []string{"GET", "POST"},
+		AllowedMethods: []string{"GET", "POST", "PUT", "DELETE"},
 		AllowedHeaders: []string{"Authorization", "Content-Type"},
+		Debug:          true,
```
[the remark this answers is not in the document]
Comment (mateuszkrasucki) [reply cmd/localAuth/main.go]: Ok

**Comment by mateuszkrasucki** [comment]
@dominikznidar all resolved, now this branch includes #36 changes so please merge first.

File: frontend/cloud/src/containers/locations/detail.js [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    }
+
+    determineState(props) {
+        let loading = (!props.location && props.locationID !== "new") || props.locationLoading || props.canEdit === undefined || props.canSee === undefined || props.validationsLoading || !props.countries || props.codesLoading
+        this.setState({ loading: loading })
+
+        if (props.location) {
+            let manager = _.clone(props.location.manager)
+
+            this.setState({ name: props.location.name })
```
Comment (matevzmihalic) [inline frontend/cloud/src/containers/locations/detail.js]: Do this in just one `setState`. And you can also shorten those ifs:
```
this.setState({
    name: props.location.name,
    capacity: props.location.capacity || "",
    ....
```

**Comment by mateuszkrasucki** [comment]
@matevzmihalic prettier done (and installed, thanks, was missing some auto formatter but didn't look for it, lol); will condense setState in a moment.

**Pull request merged**